### PR TITLE
payment-method

### PR DESCRIPTION
**Pull request by saroopchand**
- Create Custom Payment Module Structure.
- Payment Method Setting showing In Admin.
- Payment Method after Enable showing In Frontend on checkout page.
- Following Core Module : vendor/magento/module-authorizenet/

### CODE REVIEW

File: code/Progos/Custompayment/Model/PaymentMethod.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace Progos\Custompayment\Model;
+
+/**
+ * Progos Custom Payment Method Model
```
Comment (sandermangel) [inline code/Progos/Custompayment/Model/PaymentMethod.php:6]: For future reference, please describe per class why it is needed and what it does

File: code/Progos/Custompayment/etc/config.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../../Store/etc/config.xsd">
+    <default>
+        <payment>
+            <custompayment>
+                <payment_action>authorize</payment_action><!-- You can use another methor like capture  -->
```
Comment (sandermangel) [inline code/Progos/Custompayment/etc/config.xml:6]: Why only Authorize. What kind of actions would you expect this method to have?

File: code/Progos/Custompayment/etc/module.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noTestSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="Progos_Custompayment" setup_version="2.0.0" active="true">
```
Comment (sandermangel) [inline code/Progos/Custompayment/etc/module.xml:3]: Why did you start at 2.0?

File: code/Progos/Custompayment/Model/PaymentMethod.php (hunk 1 above)
Comment (saroopchand) [reply code/Progos/Custompayment/Model/PaymentMethod.php:6]: Yes sure, I will do proper commenting on code. First I just tried to build a structure where I can full fill requirement.

File: code/Progos/Custompayment/etc/config.xml (hunk 2 above)
Comment (saroopchand) [reply code/Progos/Custompayment/etc/config.xml:6]: I just go through the different extensions by Magento and blogs, they are giving me only two payment_method , authorize and capture. So I just first to authorize.

File: code/Progos/Custompayment/etc/module.xml (hunk 3 above)
Comment (saroopchand) [reply code/Progos/Custompayment/etc/module.xml:3]: I copied this code from an module and did not notice this version. I will review version and start this with 1.0.0.

**Pull request merged**